src/measure_theory/card_measurable_space.lean

```diff
@@ -0,0 +1,135 @@
+/-
+Copyright (c) 2022 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+import measure_theory.measurable_space_def
+import set_theory.continuum
+import set_theory.cofinality
+
+/-!
+# Cardinal of sigma-algebras
+
+If a sigma-algebra is generated by a set of sets `s` whose cardinality is at most the continuum,
+then the sigma-algebra satisfies the same cardinality bound. This is stated in
+`measurable_space.cardinal_generate_measurable_le` and
+`measurable_space.cardinal_measurable_set_le`.
+
+For the proof, we rely on an explicit inductive construction of the sigma-algebra generated by
+`s` (instead of the inductive predicate `generate_measurable`). This transfinite inductive
+construction is parameterized by an ordinal `< ω₁`, and the cardinality bound is preserved along
+each step of the construction.
+-/
+
+universe u
+variables {α : Type u}
+
+open_locale cardinal
+open cardinal set
+
+local notation a `<₁` b := (aleph 1 : cardinal.{u}).ord.out.r a b
+local notation `ω₁`:= (aleph 1 : cardinal.{u}).ord.out.α
+
+namespace measurable_space
+
+/-- Transfinite induction construction of the sigma-algebra generated by a set of sets `s`. At each
+step, we add all elements of `s`, the empty set, the complements of already constructed sets, and
+countable unions of already constructed sets. We index this construction by an ordinal `< ω₁`, as
+this will be enough to generate all sets in the sigma-algebra. -/
+def generate_measurable_rec (s : set (set α)) : ω₁ → set (set α)
+| i := let S := ⋃ j : {j // j <₁ i}, generate_measurable_rec j.1 in
+    s ∪ {∅} ∪ compl '' S ∪ (set.range (λ (f : ℕ → S), ⋃ n, (f n).1))
+using_well_founded {dec_tac := `[exact j.2]}
+
+/-- At each step of the inductive construction, the cardinality bound `≤ 𝔠` is preserved. -/
+lemma cardinal_generate_measurable_rec_le_aux {s : set (set α)} (hs : #s ≤ 𝔠) (i : ω₁) :
```

Comment:
Lemmas about cardinals in mathlib generally use `mk` instead of `cardinal`. Further, `aux` doesn't really clarify what this result is about. Perhaps this could be called `mk_generate_measurable_rec_le`, and the next result could be `mk_Union_generate_measurable_rec_le`?

Reply:
`mk` is a very general term in mathlib, meaning "create something out of something else". When you are in the cardinal namespace, it will mean `cardinal.mk`, but otherwise the meaning is completely ambiguous, so I'd rather stick to cardinal here.

Reply:
That's fair, though my remark about `aux` still applies.

Reply:
Yes, I've changed the names according to your comment.